src/payment/strategies/affirm/affirm-payment-strategy.spec.ts

```diff
@@ -0,0 +1,254 @@
+            };
+        });
+        window.affirm = getAffirmScriptMock();
+        window.affirm.checkout.open = jest.fn();
+        orderRequestSender = new OrderRequestSender(requestSender);
+        store = createCheckoutStore({
+            checkout: getCheckoutState(),
+            customer: getCustomerState(),
+            config: getConfigState(),
+            cart: getCartState(),
+            paymentMethods: getPaymentMethodsState(),
+            consignments: getConsignmentsState(),
+            billingAddress: getBillingAddressState(),
+        });
+        checkoutRequestSender = new CheckoutRequestSender(requestSender);
+        checkoutValidator = new CheckoutValidator(checkoutRequestSender);
+        orderActionCreator = new OrderActionCreator(orderRequestSender, checkoutValidator);
+        paymentActionCreator = new PaymentActionCreator(
+            new PaymentRequestSender(createPaymentClient()),
+            orderActionCreator
+        );
+        remoteCheckoutActionCreator = new RemoteCheckoutActionCreator(
+            new RemoteCheckoutRequestSender(requestSender)
+        );
+        strategy = new AffirmPaymentStrategy(
+            store,
+            orderActionCreator,
+            paymentActionCreator,
+            remoteCheckoutActionCreator
+        );
+
+        paymentMethod = getAffirm();
+
+        payload = merge({}, getOrderRequestBody(), {
+            payment: {
+                methodId: paymentMethod.id,
+                gatewayId: paymentMethod.gateway,
+            },
+        });
+
+        initializePaymentAction = of(createAction(RemoteCheckoutActionType.InitializeRemotePaymentRequested));
+        loadRemoteSettingsAction = of(createAction(
+            RemoteCheckoutActionType.LoadRemoteSettingsSucceeded,
+            { useStoreCredit: false },
+            { methodId: paymentMethod.id }
+        ));
+        submitOrderAction = of(createAction(OrderActionType.SubmitOrderRequested));
+        submitPaymentAction = of(createAction(PaymentActionType.SubmitPaymentRequested));
+
+        jest.spyOn(store, 'dispatch');
+
+        jest.spyOn(orderActionCreator, 'submitOrder')
+            .mockReturnValue(submitOrderAction);
+
+        jest.spyOn(remoteCheckoutActionCreator, 'initializePayment')
+            .mockReturnValue(initializePaymentAction);
+
+        jest.spyOn(remoteCheckoutActionCreator, 'loadSettings')
+            .mockReturnValue(loadRemoteSettingsAction);
+
+        jest.spyOn(paymentActionCreator, 'submitPayment')
+            .mockReturnValue(submitPaymentAction);
+
+        jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod')
+            .mockReturnValue(paymentMethod);
+
+        payload = merge({}, getOrderRequestBody(), {
+            payment: {
+                methodId: paymentMethod.id,
+                gatewayId: paymentMethod.gateway,
+            },
+        });
+    });
+
+    describe('#initialize()', () => {
```

Comment:
There's a whole bunch of logic around preparing the data required for Affirm script to initialize - in fact most of the code is for that purpose. How come we don't see any tests around it?